clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroStoreClient.java

```diff
@@ -270,6 +275,27 @@ protected RecordSerializer<K> getKeySerializerForRequest() {
         return getKeySerializerWithRetryWithShortInterval();
       } catch (Exception e) {
         whetherStoreInitTriggeredByRequestFail = true;
+        // Kick off an async thread to keep retrying
+        asyncStoreInitThread = new Thread(() -> {
+          while (true) {
+            try {
+              getKeySerializerWithRetryWithShortInterval();
+              whetherStoreInitTriggeredByRequestFail = false;
+              LOGGER.info("Successfully init store client by async store init thread");
+              break;
+            } catch (Exception ee) {
+              LOGGER
+                  .error("Received exception while trying to init store client asynchronously, will keep retrying", ee);
+              try {
+                Thread.sleep(asyncStoreInitSleepIntervalMs);
+              } catch (InterruptedException ie) {
+                Thread.currentThread().interrupt();
+                throw new VeniceClientException("Async store init thread got interrupted");
```

Comment:
Should just exit from the loop (to stop the thread), there is no one to catch and handle that `VeniceClientException`

Reply:
I guess the outcome is same and it is little more flexible in case we want to catch the exception in the future and log it.

Reply:
But it's already a top-level function in a thread, who is suppose to catch it?

Reply:
So the idea discussed here is to log the interrupted message and break the logic, right?

Reply:
Updated the PR to log the interrupted message and break the loop.